scalding-core/src/main/scala/com/twitter/scalding/Config.scala

```diff
@@ -51,6 +51,20 @@ trait Config extends Serializable {
     }
 
   /**
+   * Add files to be localized to the config. Intended to be used by user code.
+   * @param cachedFiles CachedFiles to be added
+   * @return new Config with cached files
+   */
+  def addDistributedCacheFiles(cachedFiles: CachedFile*): Config =
```

Comment:
`addDistributedCacheFile(uri: URI, config: Config)` below makes sense to me, but this one I don't quite follow. If someone already has a CachedFile, that's only possible if they provided DistributedCacheFile with a mode, and that mode has already been mutated. It seems like we don't really need this method at all? 

I think 2 instance methods (not static ones) on Config should be enough:

`def addDistributedCacheFiles(uris: URII): Config`
`def addDistributedCacheFiles(files: File*): Config`

Execution users will likely not have a CachedFile to begin with, as they won't have a mode to give to DistributedCacheFile right?